storage-provider/server/src/storage.rs

```diff
@@ -160,19 +165,20 @@ fn configure_router(state: Arc<StorageServerState>) -> Router {
 /// ```
 #[tracing::instrument(skip_all, fields(cid))]
 async fn upload(
-    ref s @ State(ref state): State<Arc<StorageServerState>>,
+    State(ref state): State<Arc<StorageServerState>>,
     Path(cid): Path<String>,
     request: Request,
 ) -> Result<String, (StatusCode, String)> {
+    // Parse the provided CID.
     let deal_cid = cid::Cid::from_str(&cid).map_err(|err| {
         tracing::error!(cid, "failed to parse cid");
         (StatusCode::BAD_REQUEST, err.to_string())
     })?;
 
+    // Use deal_db (we need it now, so we clone it)
```

Comment:
This comment adds no value. I don't get it, you're removing contextual comments that provide insight to why things are done the way they are, but add comments like this where it's very obvious what's going on.